### DISCUSSION TITLE
Wazuh Agent connection Error

### DISCUSSION THREAD

**Discussion by Jassecteam** [Security-Onion-Solutions/securityonion#2575] [Unsupported Versions]
While connecting with the Wazuh Agent it shows the following error .We have provided the Ip address of manager and authentication key. What could be the reason

2021/01/08 17:42:41 ossec-agent: INFO: (6003): Monitoring directory/file: 'c:\windows\sysnative', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | attributes | scheduled'.
2021/01/08 17:42:41 ossec-agent: INFO: (6003): Monitoring directory/file: 'c:\windows\sysnative\drivers\etc', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | attributes | scheduled'.
2021/01/08 17:42:41 ossec-agent: INFO: (6003): Monitoring directory/file: 'c:\windows\sysnative\wbem', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | attributes | scheduled'.
2021/01/08 17:42:41 ossec-agent: INFO: (6003): Monitoring directory/file: 'c:\windows\sysnative\windowspowershell\v1.0', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | attributes | scheduled'.
2021/01/08 17:42:41 ossec-agent: INFO: (6003): Monitoring directory/file: 'c:\windows\system32', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | attributes | scheduled'.
2021/01/08 17:42:41 ossec-agent: INFO: (6003): Monitoring directory/file: 'c:\windows\system32\drivers\etc', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | attributes | scheduled'.
2021/01/08 17:42:41 ossec-agent: INFO: (6003): Monitoring directory/file: 'c:\windows\system32\wbem', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | attributes | scheduled'.
2021/01/08 17:42:41 ossec-agent: INFO: (6003): Monitoring directory/file: 'c:\windows\system32\windowspowershell\v1.0', with options 'size | permissions | owner | group | mtime | inode | hash_md5 | hash_sha1 | hash_sha256 | attributes | scheduled'.
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'file' entry 'c:\programdata\microsoft\windows\start menu\programs\startup\desktop.ini'
2021/01/08 17:42:41 ossec-agent: INFO: (6207): Ignore 'file' sregex '.log$|.htm$|.jpg$|.png$|.chm$|.pnf$|.evtx$'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\Security\Policy\Secrets'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\Security\SAM\Domains\Account\Users'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\AppCs'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\DHCP'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\IPTLSIn'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\IPTLSOut'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\RPC-EPMap'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\MpsSvc\Parameters\PortKeywords\Teredo'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\PolicyAgent\Parameters\Cache'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\RunOnceEx'
2021/01/08 17:42:41 ossec-agent: INFO: (6206): Ignore 'registry' entry 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Services\ADOVMPPackage\Final'
2021/01/08 17:42:41 ossec-agent: INFO: (6207): Ignore 'registry' sregex '\Enum$'
2021/01/08 17:42:41 ossec-agent: INFO: Started (pid: 10744).
**2021/01/08 17:43:46 ossec-agent: INFO: Closing connection to server (xx.xx.xx.xx:1514/udp).
2021/01/08 17:43:46 ossec-agent: INFO: Trying to connect to server (xx.xx.xx.xx:1514/udp).
2021/01/08 17:44:51 ossec-agent: INFO: Closing connection to server (xx.xx.xx.xx:1514/udp).
2021/01/08 17:44:51 ossec-agent: INFO: Trying to connect to server (xx.xx.xx.xx:1514/udp).**

**Comment by MDCAllan** [comment]
Out of curiosity, are your agents configured on a manager or on a forward node?

We're having the same problem, and it appears to be because the yaml file that defines roles for hosts/ports doesn't have anything defined for the forward node, so so-allow doesn't actually open ports on the forward node for any services.

**Reply by facyber** [reply]
Same issue here, with Heavy Node and Manager node. BY logic it should be on heavy node in my case because its doing all the work sort of say, but so-allow does not work because there is no file `/opt/so/saltstack/local/salt/firewall/hostgroups.local.yaml`.

There is this thread #2724 but I couldn't understand whether it should be done on manager or sensor/heavy node. On manager node is not logical, but then again it is not possible on heavy node because of the issue I mentioned.

**Comment by innovate-support** [comment]
We seem to have finally come up with a solution to this lingering issue.
https://github.com/Security-Onion-Solutions/securityonion/discussions/2455#discussioncomment-364087
